### PR TITLE
Fixes for segment mapping error

### PR DESCRIPTION
**Pull request by gmbr3**
Required fix check by @stormc

### CODE REVIEW

File: gnuefi/crt0-efi-loongarch64.S [hunk 1]
```diff
@@ -45,12 +45,13 @@ _start:
...
 
  	.data
-dummy:	.4byte	0
+dummy0:	.4byte	0
+dummy1:	.4byte	0
 
 #define IMAGE_REL_ABSOLUTE	0
- 	.section .reloc, "a"
+ 	.section .reloc, "a", %progbits
 label1:
```
Comment (gmbr3) [inline gnuefi/crt0-efi-loongarch64.S:53]: label1 needs removing

File: gnuefi/crt0-efi-ia32-local.S [hunk 2]
```diff
@@ -163,11 +163,12 @@ _start:
...
+dummy0:	.4byte	0
+dummy1:	.4byte	0
 
 #define IMAGE_REL_ABSOLUTE	0
- 	.section .reloc
- 	.4byte	dummy					// Page RVA
+ 	.section .areloc, "a", %progbits
+ 	.4byte	dummy1 - dummy0					// Page RVA
 	.4byte	12					// Block Size (2*4+2*2), must be aligned by 32 Bits
 	.2byte	(IMAGE_REL_ABSOLUTE<<12) +  0		// reloc for dummy
```
Comment (gmbr3) [inline gnuefi/crt0-efi-ia32-local.S:173]: These can be thinned to only dummy0 I think

File: gnuefi/crt0-efi-ia32-local.S [hunk 3]
```diff
@@ -163,11 +163,12 @@ _start:
...
  	.data
-dummy:	.4byte	0
+dummy0:	.4byte	0
+dummy1:	.4byte	0
 
 #define IMAGE_REL_ABSOLUTE	0
- 	.section .reloc
- 	.4byte	dummy					// Page RVA
+ 	.section .areloc, "a", %progbits
+ 	.4byte	dummy1 - dummy0					// Page RVA
```
Comment (gmbr3) [inline gnuefi/crt0-efi-ia32-local.S:171]: This looks wrong it's supposed to be Page RVA even if it gets ignored anyways
Should be dummy0??

Comment (gmbr3) [reply gnuefi/crt0-efi-ia32-local.S]: Likely wrong but assembly symbols don't calculate

File: gnuefi/crt0-efi-ia32-local.S (hunk 2 above)
Comment (gmbr3) [reply gnuefi/crt0-efi-ia32-local.S:172]: Same issue

File: gnuefi/crt0-efi-loongarch64.S (hunk 1 above)
Comment (gmbr3) [reply gnuefi/crt0-efi-loongarch64.S]: done

**Comment by jan-kiszka** [comment]
FWIW, this seems to resolve the corruptions we (including @stormc) were seeing with EFI Boot Guard, see also https://gitlab.com/cip-project/cip-core/isar-cip-core/-/issues/114. I've just rebuilt gnu-efi master against trixie, and isar-cip-core boots qemu-arm again. Hope we can see a new release soon so that we can poke Debian to update (and also move away from the ambiguous 3.0.18 release).

**Comment by jan-kiszka** [comment]
Turned out it was rather patch 24a4cd0e5653fd84b004c00c808c45cc3fb7a7e2 that resolved the armhf issue. Will open a Debian bug while waiting for 3.0.19.

**Pull request merged**